tests/src/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -103,14 +126,60 @@ public class CoreclrTestWrapperLib
 
         static bool CollectCrashDump(Process process, string path)
         {
-            using (var crashDump = File.OpenWrite(path))
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                using (var crashDump = File.OpenWrite(path))
+                {
+                    var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
+                    return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+                }
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                string coreRoot = Environment.GetEnvironmentVariable("CORE_ROOT");
+                ProcessStartInfo createdumpInfo = new ProcessStartInfo("sudo");
+                createdumpInfo.Arguments = $"{Path.Combine(coreRoot, "createdump")} --name \"{path}\" {process.Id} -h";
+                Process createdump = Process.Start(createdumpInfo);
+                return createdump.WaitForExit(DEFAULT_TIMEOUT) && createdump.ExitCode == 0;
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
             {
-                var flags = DbgHelp.MiniDumpType.MiniDumpWithFullMemory | DbgHelp.MiniDumpType.MiniDumpIgnoreInaccessibleMemory;
-                return DbgHelp.MiniDumpWriteDump(process.Handle, process.Id, crashDump.SafeFileHandle, flags, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+                int pid = process.Id;
+                Console.WriteLine($"Aborting process {pid} to generate dump");
```

Comment:
Why not make this consistent message across different OSs? Seems like an easy thing to grep for in logs.

Reply:
We don't abort processes on timeout on other platforms. We just create a dump of the process.